examples/nn/cnn_customop.py

```diff
@@ -0,0 +1,124 @@
+"""Convolution Neural Network example with both MinPy ndarray and MXNet symbol."""
+import sys
+import argparse
+
+import minpy
+import minpy.numpy as np
+import mxnet as mx
+from minpy.nn.io import NDArrayIter
+# Can also use MXNet IO here
+# from mxnet.io import NDArrayIter
+from minpy.core import Function
+from minpy.nn import layers
+from minpy.nn.model import ModelBase
+from minpy.nn.solver import Solver
+from examples.utils.data_utils import get_CIFAR10_data
+from minpy.primitive import CustomOp
+
+# Please uncomment following if you have GPU-enabled MXNet installed.
+#from minpy.context import set_context, gpu
+#set_context(gpu(0)) # set the global context as gpu(0)
+
+batch_size=128
+input_size=(3, 32, 32)
+flattened_input_size=3 * 32 * 32
+hidden_size=512
+num_classes=10
+
+class softmax_loss(CustomOp):
+    def forward(self, x, y):
+        N = x.shape[0]
+        C = x.shape[1]
+        if len(y.shape) == 1:
+            #convert it to one hot encoding
+            onehot_y = np.zeros([N, C])
+            np.onehot_encode(y, onehot_y)
+        else:
+            onehot_y = y
+        probs = x - np.max(x, axis=1, keepdims=True)
+        loss = -np.sum(probs * onehot_y) / N
+        loss += np.sum(np.log(np.sum(np.exp(probs), axis=1, keepdims=True))) / N
+        return loss
```

Comment:
Current solution cannot share info other than `ans, *args, **kwargs`, which is not expected as I mentioned in 1.